pkg/controller/global/image/controller.go

```diff
@@ -0,0 +1,261 @@
+package image
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/minio/minio-go/v6"
+	apisv1alpha1 "github.com/rancher/vm/pkg/apis/vm.cattle.io/v1alpha1"
+	"github.com/rancher/vm/pkg/config"
+	"github.com/rancher/vm/pkg/generated/controllers/vm.cattle.io/v1alpha1"
+	"github.com/rancher/vm/pkg/util"
+	"github.com/rancher/wrangler-api/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	controllerAgentName = "vm-image-controller"
+)
+
+var (
+	syncProgressInterval = 2 * time.Second
+	importIdleTimeout    = 5 * time.Minute
+)
+
+func RegisterController(ctx context.Context, scaled *config.Scaled) {
+	images := scaled.VMFactory.Vm().V1alpha1().Image()
+	controller := &Handler{
+		images:     images,
+		imageCache: images.Cache(),
+		//FIXME panic in framework ATM
+		//secretCache: scaled.CoreFactory.Core().V1().Secret().Cache(),
```

Comment:
What does this comment mean?

Reply:
Nil pointer dereference panic in `github.com/rancher/dynamiclistener@v0.2.1-0.20200213165308-111c5b43e932/storage/kubernetes/controller.go:144` when secret cache is used. I'm removing the secretRef handling for now.